src/super_gradients/training/processing/processing.py

```diff
@@ -668,6 +684,202 @@ def resizes_image(self) -> bool:
+
+    def postprocess_predictions(self, predictions: SegmentationPrediction, metadata: SegmentationResizeMetadata) -> SegmentationPrediction:
+        predictions.segmentation_map = _rescale_image(
+            predictions.segmentation_map.astype(np.uint8), target_shape=metadata.original_shape, interpolation_method=cv2.INTER_NEAREST
+        )
+        predictions.segmentation_map = predictions.segmentation_map.astype(np.int64)
+        return predictions
+
+    def get_equivalent_photometric_module(self) -> Optional[nn.Module]:
+        return None
+
+    @property
+    def resizes_image(self) -> bool:
+        return True
+
+
+@register_processing(Processings.SegmentationPadShortToCropSize)
+class SegmentationPadShortToCropSize(Processing):
+    """
+        Pads image to 'crop_size'.
+        Should be called only after "SegRescale" or "SegRandomRescale" in augmentations pipeline.
+
+        :param crop_size:   Tuple of (width, height) for the final crop size, if is scalar size is a square (crop_size, crop_size)
+    =    :param fill_image:  Grey value to fill image padded background.
+    """
+
+    def __init__(self, crop_size: Union[float, Tuple, List], fill_image: Union[int, Tuple, List]):
+        self.crop_size = crop_size
+        self.fill_image = fill_image
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, DetectionPadToSizeMetadata]:
+        # pad images from center symmetrically
+        output_shape = max(image.shape[0], self.crop_size[0]), max(image.shape[1], self.crop_size[1])
+        padding_coordinates = _get_center_padding_coordinates(input_shape=image.shape, output_shape=output_shape)
+        padded_image = _pad_image(image=image, padding_coordinates=padding_coordinates, pad_value=self.fill_image)
+
+        return padded_image, DetectionPadToSizeMetadata(padding_coordinates=padding_coordinates)
+
+    def postprocess_predictions(self, predictions: SegmentationPrediction, metadata: DetectionPadToSizeMetadata) -> SegmentationPrediction:
+        predictions.segmentation_map = predictions.segmentation_map[
+            metadata.padding_coordinates.top : predictions.segmentation_map_shape[0] - metadata.padding_coordinates.bottom,
+            metadata.padding_coordinates.left : predictions.segmentation_map_shape[1] - metadata.padding_coordinates.right,
+        ]
+        return predictions
+
+    def get_equivalent_photometric_module(self) -> Optional[nn.Module]:
+        return None
+
+    @property
+    def resizes_image(self) -> bool:
+        return True
+
+
+@register_processing(Processings.SegmentationPadToDivisible)
+class SegmentationPadToDivisible(Processing):
+    """
+        Pads image to a size divisible by the defined parameter.
+
+        :param divisible_value:   the divisible value, new image size is an int multiplication of this number
+    =    :param fill_image:  Grey value to fill image padded background.
```

Comment:
Too much indented and include a typo (`=`)

Reply:
Thanks for spotting! Fixed in 24538f292d59682d4a54ab82db048b62c62dddeb